### ISSUE TITLE
No SSD optimization

### ISSUE THREAD

**Issue by lil5** [nazar-pc/just-backup-btrfs#5]
From what I read I can't find an option to optimize for SSD's

**Comment by nazar-pc** [comment]
What exactly SSD optimization are you talking about?

**Comment by lil5** [comment]
Is Btrfs optimized for SSD?

There are some optimizations for SSD drives, and you can enable them by mounting with -o ssd. As of 2.6.31-rc1, this mount option will be enabled if Btrfs is able to detect non-rotating storage.

Note that before 4.14 the ssd mount option has a negative impact on usability and lifetime of modern SSDs which have a FTL (Flash Translation Layer). See the ssd section in Gotchas for more information.

Note that -o ssd will not enable TRIM/discard.

https://btrfs.wiki.kernel.org/index.php/FAQ#Is_Btrfs_optimized_for_SSD.3F

**Comment by nazar-pc** [comment]
Yes, BTRFS has special mount option for SSD optimization.

However, Just backup btrfs additionally mounts partitions into separate location in `/tmp` only because it needs this in order to work properly and while doing so it uses exactly the same mount options as those that were used for mounting originally.

Here is the relevant piece of code: https://github.com/nazar-pc/just-backup-btrfs/blob/a108705f442a29fbb20fb367b4d904932d9dd7b2/just-backup-btrfs#L207-L242

In other words, if `destination_other_partition` is on SSD and you've used mount options for optimizations - Just backup btrfs will naturally use the same options too.

**Comment by nazar-pc** [comment]
Also `destination_other_partition` will likely be HDD anyway while `destination_within_partition` is not even re-mounted in any way at all.

I think this should answer your question pretty well.

**Comment by mikhailnov** [comment]
In modern kernels btrfs automatically detects SSDs and applies optimizations, so ther is no need to pass special mount options.
